test/TestCudaUmpireResource.cpp

```diff
@@ -0,0 +1,71 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
+// See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "care/config.h"
+
+#if defined(CARE_GPUCC)
+
+// other library headers
+#include "gtest/gtest.h"
+#include "umpire/strategy/QuickPool.hpp"
+
+// care headers
+#include "care/CudaUmpireResource.h"
+#include "care/detail/test_utils.h"
+
+GPU_TEST(CudaUmpireResource, gpu_initialization) {
+   init_care_for_testing();
+}
+
+GPU_TEST(CudaUmpireResource, DefaultConstructor)
+{
+   care::CudaUmpireResource resource;
+}
+
+GPU_TEST(CudaUmpireResource, AllocatorConstructor)
+{
+   auto& rm = umpire::ResourceManager::getInstance();
```

Comment:
@neelakausik, essentially you will need to create a device pool, a pinned pool, and a unified memory pool per thread, then use those to create a resource per thread. The call to init_care_for_testing() above creates a GPU wide pool called "DEVICE_POOL" and another pool called "PINNED_POOL". Those should be the basis for the per thread device and pinned memory pools. On line 53 I create a GPU wide unified memory pool called "UM_POOL". You will need to create that manually in whatever test cases you have, then have that allocator be the basis for the per thread unified memory pool.

Reply:
You should test out this functionality in standalone CARE. Write some tests and benchmarks that involve RAJA reducers, scans, and CARE loops.